grails-app/assets/javascripts/charts2.js

```diff
@@ -103,12 +168,12 @@ function cleanUp(chartOptions) {
 \*********************************************************************/
 function drawFacetCharts(data, chartOptions) {
     // check that we have results
-    if (data.length == 0 || data.totalRecords == undefined || data.totalRecords == 0) {
+    if(!data.length || !data.totalRecords || !data.totalRecords) {
```

Comment:
``!data.totalRecords`` is twice. Also, we usually don't test for ``0`` implicitly. Something like this:

```javascript
data.length === 0 || !data.totalRecords
```